client/db_test.go

```diff
@@ -288,3 +289,31 @@ func TestDebugName(t *testing.T) {
 		return nil
 	})
 }
+
+func TestCommonMethods(t *testing.T) {
+	methods := []string{
```

Comment:
A test based on whitelisting the methods that are expected to exist everywhere won't help much - it won't catch new methods like GetProto unless you know to add them to the list. I think we need a blacklist instead, of methods that are allowed to differ between the different types.